src/engine/source/base/include/base/utils/threadEventDispatcher.hpp

```diff
@@ -30,22 +44,19 @@ template<typename T,
 class TThreadEventDispatcher
 {
 public:
-    explicit TThreadEventDispatcher(Functor functor,
-                                    const std::string& dbPath,
-                                    const uint64_t bulkSize = 1,
-                                    const size_t maxQueueSize = UNLIMITED_QUEUE_SIZE,
-                                    const uint8_t numberOfThreads = 1)
+    explicit TThreadEventDispatcher(
+        Functor functor,
+        const std::string& dbPath,
+        const uint64_t bulkSize = 1,
+        const size_t maxQueueSize = UNLIMITED_QUEUE_SIZE,
+        const ThreadEventDispatcherType numberOfThreads = ThreadEventDispatcherType::SINGLE_THREADED_ORDERED)
```

Comment:
Why the number of threads is of the type ThreadEventDispatcherType?
```
        const ThreadEventDispatcherType numberOfThreads = ThreadEventDispatcherType::SINGLE_THREADED_ORDERED)
```

Reply:
Changed by dispatcherType